components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/api/impl/BusinessRuleApiServiceImpl.java

```diff
@@ -30,7 +40,17 @@ public Response getDeploymentStatus(String instanceUUID
     @Override
     public Response loadBusinessRuleInstance(String instanceUUID
  ) throws NotFoundException {
-        // do some magic!
+        // Example!
+        QueryManager queryManager = new QueryManager("business.rules");
+        String q = queryManager.getQuery("RETRIEVE_BUSINESS_RULE");
+        QueryExecutor queryExecutor = new QueryExecutor();
+        try {
+            ResultSet resultSet = queryExecutor.executeRetrieveBusinessRule("SmartHomePlan");
+        } catch (BusinessRulesDatasourceException e) {
+            log.error(e.getMessage());
```

Comment:
can we provide a meaning full message with e.getMessage() ?? and any reason why we do not log the stack trace ?

Reply:
This code segment was added just as an example on how to use QueryExecutor and QueryManager. Will added more detailed logs in actual implementation.